### PR TITLE
Test summary.imugap_predict (#73)

### PR DESCRIPTION
**Pull request by WestonVoglesonger**
First targeted slice of #73: `summary.imugap_predict` (added with the draws-compatible predict refactor in #70) had no dedicated test.

## Coverage added (`tests/testthat/test-summary.R`)
- **Output shape** — one row per target observation; the `target` columns (`obs_c_id`, `loc_id`, `age`, `cohort`, `dose`) are carried through.
- **Quantile columns** — default `probs = c(.025, .5, .975)` produces `mean` + `q2_5`/`q50`/`q97_5`; custom `probs` swap them (e.g. `q10`/`q90`).
- **Correctness** — `mean` and quantiles match a manual `colMeans` / `quantile` reduction of `$draws`.
- **Error branch** — non-`imugap_predict` input is rejected.
- **Copy semantics** — `object$target` is not mutated.

Uses a small synthetic `imugap_predict` object (a `draws` matrix + `target` data.table) rather than running the sampler, so the test is fast and deterministic. Verified passing (13 assertions) against the current `main` source.

Remaining #73 items (predict edge cases, checker error branches, options/canonicalize failure paths) are left for follow-up slices.

Refs #73

### CODE REVIEW

File: tests/testthat/test-summary.R [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  draws <- matrix(runif(n_draws * n_obs), nrow = n_draws, ncol = n_obs)
+  target <- data.table::data.table(
+    obs_c_id = 1:3,
+    loc_id = c("a", "b", "c"),
+    age = c(5L, 5L, 5L),
+    cohort = c(1L, 1L, 1L),
+    dose = c(2L, 2L, 2L)
+  )
+  structure(list(draws = draws, target = target), class = "imugap_predict")
+}
```
Comment (pearsonca) [inline tests/testthat/test-summary.R:20]: this may be incomplete / incorrectly implemented in predict itself, but: we're going to want draws to be iterations x chains x variables

**Review by pearsonca — APPROVED** [review]
LGTM for predict as is - but: can you also circle back to predict and have it instead provide in the draws array style structure: https://cran.r-project.org/web/packages/posterior/vignettes/posterior.html#draws-formats (and update all the contingent items accordingly)?

It'll be a pretty common ask to trim off the burn-in period, and draws array makes that simplest.
